### PR TITLE
[OpenGL] Implement glDepthRangeIndexeddNV

### PR DESCRIPTION
**Pull request by Kelebek1**
glDepthRangeIndexeddNV allows depth buffers outside the range [0, 1], which is needed by some games. The current glDepthRangeIndexed clamps the depth to [0, 1] beforehand, which produces incorrect results with [-1, 1] depth buffers.

All credits to Rodrigo for finding this, I did nothing.

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 1]
```diff
@@ -862,9 +862,10 @@ void RasterizerOpenGL::SyncViewport() {
         if (regs.screen_y_control.y_negate != 0) {
             flip_y = !flip_y;
         }
-        const bool is_zero_to_one = regs.depth_mode == Maxwell::DepthMode::ZeroToOne;
         const GLenum origin = flip_y ? GL_UPPER_LEFT : GL_LOWER_LEFT;
-        const GLenum depth = is_zero_to_one ? GL_ZERO_TO_ONE : GL_NEGATIVE_ONE_TO_ONE;
+        const GLenum depth = regs.depth_mode == Maxwell::DepthMode::ZeroToOne
```
Comment (ReinUsesLisp) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp:866]: nit: Undo these changes to keep things in a single line.

**Review by ReinUsesLisp — COMMENTED** [review]
lgtm, just one nit.

**Review by ReinUsesLisp — APPROVED** [review]

**Review by bunnei — APPROVED** [review]
